GPSSetup/clients/GPSSetup/controllers/page/CameraGPS/onStart.js

```diff
@@ -0,0 +1,197 @@
+    var page = this;
+
+    page.data.zone = 0;  // tells the drawing tools that we are not on the zone page
+
+
+    //array for map markers
+    page.data.markers = [];
+
+    //get the map widet
+    var map = client.getWidget("DynamicMapViewer136").map;
+
+
+    page.data.camera.initializeToDefaultValues();
+    page.data.camera.copyMatchingData(parameters);
+    console.log(parameters);
+
+    let lat = page.data.camera.geo.coordinates[1];
+    let lng = page.data.camera.geo.coordinates[0];
+    var place = {lat: lat, lng: lng};
+    map.setMapTypeId('satellite');
+    map.setCenter(place);
+    map.setZoom(18);    
+
+    //var canvas = $("#callibration-canvas").get(0);
+    var canvas = document.getElementById('callibration-canvas');   
+    var ctx = canvas.getContext("2d");    
+    ctx.clearRect(0, 0, canvas.width, canvas.height);
+
+    //Little hack to load a local file into the canvas, there is a small hidden html form input tag on the page that does the file upload, the button on the page just triggers this. See the tiny square under the map.
+    var imageLoader = document.getElementById('imageLoader');
+    imageLoader.addEventListener('change', handleImage, false);
+
+    // setup the map gps markerss
+    client.data.camGeoSetup.initializeToDefaultValues();
+    var latlngbl, latlngbr, latlngtl, latlngtr;
+    console.log(page.data.camera.gpsTrack);
+
+    if (jQuery.isEmptyObject(page.data.camera.gpsTrack) === false){
+        //if (page.data.camera.gpsTrack.length > 1 ){
+        console.log("adding existing markers");
+        latlngbl = page.data.camera.gpsTrack.bottomLeftGeo;
+        addMarker(latlngbl, "bottomLeftGeo");
+
+        latlngbr = page.data.camera.gpsTrack.bottomRightGeo;
+        addMarker(latlngbr, "bottomRightGeo");
+
+        latlngtl = page.data.camera.gpsTrack.topLeftGeo;
+        addMarker(latlngtl, "topLeftGeo");
+
+        latlngtr = page.data.camera.gpsTrack.topRightGeo;
+        addMarker(latlngtr, "topRightGeo");
+
+    } else {
+        // since there are no markers set them up near the camera origin
+        console.log("adding new markers");
+        latlngbl = {lat:page.data.camera.geo.coordinates[1] - 0.0001, lng:page.data.camera.geo.coordinates[0] - 0.0001};
+        addMarker(latlngbl, "bottomLeftGeo");
+
+        latlngbr = {lat:page.data.camera.geo.coordinates[1] - 0.0001, lng:page.data.camera.geo.coordinates[0] + 0.0001};
+        addMarker(latlngbr, "bottomRightGeo");
+
+        latlngtl = {lat:page.data.camera.geo.coordinates[1] + 0.0001, lng:page.data.camera.geo.coordinates[0] - 0.0001};
+        addMarker(latlngtl, "topLeftGeo");
+
+        latlngtr = {lat:page.data.camera.geo.coordinates[1] + 0.0001, lng:page.data.camera.geo.coordinates[0] + 0.0001};
+        addMarker(latlngtr, "topRightGeo");
```

Comment:
Looks like the only difference between the if/else blocks is how the `lat___` variables are defined. Could simplify by  setting them once, and conditionally changing them if necessary.

Then, outside the if block, you only need to call `addMarker` 4 times.

Something like:

```
var latlngbl = page.data.camera.gpsTrack.bottomLeftGeo;
var latlngbr = page.data.camera.gpsTrack.bottomRightGeo;
var latlngtl = page.data.camera.gpsTrack.topLeftGeo;
var latlngtr = page.data.camera.gpsTrack.topRightGeo;

if (jQuery.isEmptyObject(page.data.camera.gpsTrack)) {
    latlngbl = {lat:page.data.camera.geo.coordinates[1] - 0.0001, lng:page.data.camera.geo.coordinates[0] - 0.0001};
    latlngbr = {lat:page.data.camera.geo.coordinates[1] - 0.0001, lng:page.data.camera.geo.coordinates[0] + 0.0001};
    latlngtl = {lat:page.data.camera.geo.coordinates[1] + 0.0001, lng:page.data.camera.geo.coordinates[0] - 0.0001};
    latlngtr = {lat:page.data.camera.geo.coordinates[1] + 0.0001, lng:page.data.camera.geo.coordinates[0] + 0.0001};
}

addMarker(latlngbl, "bottomLeftGeo");
addMarker(latlngbr, "bottomRightGeo");
addMarker(latlngtl, "topLeftGeo");
addMarker(latlngtr, "topRightGeo");
```